.github/workflows/feature-branch-auto-merge.yml

```diff
@@ -0,0 +1,34 @@
+name: Feature branch auto merge
+
+permissions: read-all
+
```

Comment:
Could you add a concurrency group here? It would be nice to ensure that merges don't happen out of order if two commits are merged very close together if possible... or maybe we should cancel if another one is happening at the same time?